File: library/src/androidTest/mock/com/mobilesolutionworks/android/app/test/util/ConfigurationChangeLatch.java
package com.mobilesolutionworks.android.app.test.util;

import android.content.ComponentCallbacks;
import android.content.res.Configuration;
import android.util.Log;

import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

/**
 * Created by yunarta on 12/3/17.
 */
public class ConfigurationChangeLatch implements ComponentCallbacks {

    private CountDownLatch mLatch;

    public ConfigurationChangeLatch() {
        mLatch = new CountDownLatch(1);
    }

    public void await() throws InterruptedException {
        mLatch.await(5, TimeUnit.SECONDS);
        reset();
    }

    public void reset() {
        mLatch = new CountDownLatch(1);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        mLatch.countDown();
    }

    @Override
    public void onLowMemory() {

    }
}

File: test-app/src/androidTest/kotlin/com/mobilesolutionworks/android/controller/test/util/ConfigurationChangeLatch.kt
package com.mobilesolutionworks.android.controller.test.util

import android.content.ComponentCallbacks
import android.content.res.Configuration

import java.util.concurrent.CountDownLatch
import java.util.concurrent.TimeUnit

/**
 * Created by yunarta on 12/3/17.
 */
class ConfigurationChangeLatch : ComponentCallbacks {

    private var mLatch: CountDownLatch? = null

    init {
        mLatch = CountDownLatch(1)
    }

    @Throws(InterruptedException::class)
    fun await() {
        mLatch!!.await(5, TimeUnit.SECONDS)
        reset()
    }

    fun reset() {
        mLatch = CountDownLatch(1)
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        mLatch!!.countDown()
    }

    override fun onLowMemory() {

    }
}
